commit 96d88f65adfbcaca153afd7d3e20d74ba379c599
Author: Brian Foster <bfoster@redhat.com>
Date:   Thu Apr 25 06:44:00 2024 -0400

    virtiofs: include a newline in sysfs tag
    
    The internal tag string doesn't contain a newline. Append one when
    emitting the tag via sysfs.
    
    [Stefan] Orthogonal to the newline issue, sysfs_emit(buf, "%s", fs->tag) is
    needed to prevent format string injection.
    
    Signed-off-by: Brian Foster <bfoster@redhat.com>
    Fixes: a8f62f50b4e4 ("virtiofs: export filesystem tags through sysfs")
    Signed-off-by: Miklos Szeredi <mszeredi@redhat.com>

diff --git a/fs/fuse/virtio_fs.c b/fs/fuse/virtio_fs.c
index 322af827a232..bb3e941b9503 100644
--- a/fs/fuse/virtio_fs.c
+++ b/fs/fuse/virtio_fs.c
@@ -170,7 +170,7 @@ static ssize_t tag_show(struct kobject *kobj,
 {
 	struct virtio_fs *fs = container_of(kobj, struct virtio_fs, kobj);
 
-	return sysfs_emit(buf, fs->tag);
+	return sysfs_emit(buf, "%s\n", fs->tag);
 }
 
 static struct kobj_attribute virtio_fs_tag_attr = __ATTR_RO(tag);